packages/smart-wallet/src/smartWallet.js

```diff
@@ -1085,14 +1103,40 @@ export const prepareSmartWallet = (baggage, shared) => {
       resultStepWatcher: {
         /**
          * @param {unknown} result
-         * @param {ResultPlan} saveResult
+         * @param {{ id?: string | number; saveResult?: ResultPlan }} opts
          */
-        onFulfilled(result, saveResult) {
-          trace('resultStepWatcher result', result);
-          this.facets.helper.saveEntry(saveResult, result);
+        onFulfilled(result, opts) {
+          trace('resultStepWatcher opts', opts);
+          if (opts.saveResult) {
+            this.facets.helper.saveEntry(opts.saveResult, result);
+          }
+          const passStyle = passStyleOf(result);
+          const { updateRecorderKit } = this.state;
+          if (opts.id) {
+            void updateRecorderKit.recorder.write({
+              updated: 'invocation',
+              id: opts.id,
+              result: {
+                ...(opts.saveResult?.name && { name: opts.saveResult.name }),
```

Comment:
FYI this is only safe because `{...''}` (an empty name) equals `{}`, but I would prefer a ternary (or `??`) depending on how we want to handle a falsy empty name.

Reply:
`??` complains about spread and object types.

if I need another round of ci, what do you think of `!!opts.saveResult?.name`